Pear/Profile/ProfileSectionTableViewCell.swift

```diff
@@ -0,0 +1,114 @@
+//
+//  ProfileSectionTableViewCell.swift
+//  Pear
+//
+//  Created by Lucy Xu on 3/13/21.
+//  Copyright © 2021 cuappdev. All rights reserved.
+//
+
+import UIKit
+import Kingfisher
+
+class ProfileSectionTableViewCell: UITableViewCell {
+
+    // MARK: - Private View Vars
+    private let titleLabel = UILabel()
+    private let itemsCollectionView = SelfSizingCollectionView()
+
+    static let reuseIdentifier = "ProfileSectionTableViewCell"
+    private var sectionType: ProfileSectionType? = nil
+    private var items: [String] = []
+
+    override init(style: UITableViewCell.CellStyle, reuseIdentifier: String?) {
+        super.init(style: style, reuseIdentifier: reuseIdentifier)
+
+        selectionStyle = .none
+        backgroundColor = .clear
+
+        titleLabel.font = .boldSystemFont(ofSize: 12)
+        titleLabel.textColor = .greenGray
+        contentView.addSubview(titleLabel)
+
+        itemsCollectionView.frame = CGRect(x: 0, y: 0, width: frame.width, height: 0)
+        itemsCollectionView.dataSource = self
+        itemsCollectionView.delegate = self
+        itemsCollectionView.backgroundColor = .clear
+        itemsCollectionView.register(
+            ProfilePillCollectionViewCell.self,
+            forCellWithReuseIdentifier: ProfilePillCollectionViewCell.reuseIdentifier
+        )
+        itemsCollectionView.layoutIfNeeded()
+        contentView.addSubview(itemsCollectionView)
+
+        setupConstraints()
+    }
+
+    private func setupConstraints() {
+        titleLabel.snp.makeConstraints { make in
+            make.top.leading.trailing.equalToSuperview().offset(20)
+            make.height.equalTo(15)
+        }
+
+        itemsCollectionView.snp.makeConstraints { make in
+            make.leading.trailing.equalTo(titleLabel)
+            make.top.equalTo(titleLabel.snp.bottom).offset(8)
+            make.bottom.equalToSuperview().inset(10)
+        }
+    }
+
+    func configure(for user: User, type: ProfileSectionType) {
+        titleLabel.text = type.getTitle(for: user)
+        sectionType = type
+        switch type {
+        case .groups:
+            items = user.groups
+        case .interests:
+            items = user.interests
+        default:
+            break
+        }
+        itemsCollectionView.reloadData()
+        itemsCollectionView.sizeToFit()
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+}
+
+extension ProfileSectionTableViewCell: UICollectionViewDataSource {
+}
+
+extension ProfileSectionTableViewCell: UICollectionViewDelegate {
```

Comment:
I don't think you actually implement any delegate methods. In that case, you shouldn't extend the delegate.